### PR TITLE
Bug 1656649 - Add Fx Reality PC

### PR DESCRIPTION
**Pull request by fbertsch**
Is Glean a dependency for the C# apps?

### CODE REVIEW

**Comment by Dexterp37** [comment]
> 
> 
> Is Glean a dependency for the C# apps?

Yes, Glean is, but *not* Glean Android. Should we split [this](https://github.com/mozilla/probe-scraper/blob/5c9d656fb42c5d9f02cf85d26d156f221c9e329e/repositories.yaml#L1) in two? `glean` and `glean_android`. With `glean_android` having a dependency on `glean`? @mdboom @fbertsch ?

**Comment by fbertsch** [comment]
@Dexterp37 it sounds like that is the correct way to approach this, however all dependencies have to be explicitly defined, so we would need to add both `glean` and `glean_android` for android apps.

**Comment by Dexterp37** [comment]
> 
> 
> @Dexterp37 it sounds like that is the correct way to approach this, however all dependencies have to be explicitly defined, so we would need to add both `glean` and `glean_android` for android apps.

I see. Maybe `glean_core` and `glean_android` then, but that's indeed the way forward.

**Comment by mdboom** [comment]
> @Dexterp37 it sounds like that is the correct way to approach this, however all dependencies have to be explicitly defined, so we would need to add both `glean` and `glean_android` for android apps.

It will be hard to remember, but I think this is the correct approach.

**Comment by fbertsch** [comment]
Alternatively, we could have `glean-android` include both those metrics files. Then we would only need to include the one dependency for Android apps. Thoughts?

**Comment by Dexterp37** [comment]
> 
> 
> Alternatively, we could have `glean-android` include both those metrics files. Then we would only need to include the one dependency for Android apps. Thoughts?

Oh, that's even better :) We'll also need a comment section at the top explaining why. With that, I think we should be ok.

**Comment by fbertsch** [comment]
Great. I'll add that to this PR.

**Comment by fbertsch** [comment]
@Dexterp37 @mdboom follow-up, will Glean then also be a dependency for the Python apps?

**Comment by mdboom** [comment]
Yeah -- I'm wondering a bit why it's not already a dependency for iOS and Python apps. This would mainly impact the error metrics.  Looking at those schemas/tables now to confirm/deny my worst fears...

**Comment by Dexterp37** [comment]
> 
> 
> @Dexterp37 @mdboom follow-up, will Glean then also be a dependency for the Python apps?

Yes.

**Comment by mdboom** [comment]
> Yeah -- I'm wondering a bit why it's not already a dependency for iOS and Python apps. This would mainly impact the error metrics. Looking at those schemas/tables now to confirm/deny my worst fears...

So -- somehow the Glean core metrics are ending up in the tables for the iOS apps, even though it's not a dependency there. We should at least remind ourselves and understand why that is first.

**Comment by fbertsch** [comment]
> So -- somehow the Glean core metrics are ending up in the tables for the iOS apps, even though it's not a dependency there. We should at least remind ourselves and understand why that is first.

@mdboom is there a bug or issue filed for this? This seems important to track down.

I'll summarize this in https://github.com/mozilla/probe-scraper/issues/185 and assign that to myself.

**Comment by fbertsch** [comment]
Now fixes #185.

File: repositories.yaml [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+    - frank@mozilla.com
+    - mdroettboom@mozilla.com
+  url: 'https://github.com/mozilla/glean'
+  branch: main
+  metrics_files:
+    - 'glean-core/metrics.yaml'
+  ping_files:
+    - 'glean-core/pings.yaml'
+  library_names:
+    - org.mozilla.glean-core
```
Comment (Dexterp37) [inline repositories.yaml:13]: nit: is this required to be `org.mozilla.glean-core` (android specific?)? Can this be `glean-core` or `lib-glean-core`? We don't need to change it, just curious.

File: repositories.yaml [hunk 2]
```diff
@@ -12,6 +25,18 @@ glean:
     - 'glean-core/pings.yaml'
   library_names:
     - org.mozilla.components:service-glean
+glean-deprecated:
```
Comment (Dexterp37) [inline repositories.yaml:28]: Why are we leaving this around?

File: repositories.yaml (hunk 1 above)
Comment (fbertsch) [reply repositories.yaml:13]: Nope, it can be any valid string, really. Would you prefer `glean-core`?

File: repositories.yaml (hunk 2 above)
Comment (fbertsch) [reply repositories.yaml:28]: There's some description in this commit: https://github.com/mozilla/probe-scraper/pull/217/commits/1db46bea9dca73a55c384b10b25536cb43203af6

File: repositories.yaml (hunk 1 above)
Comment (Dexterp37) [reply repositories.yaml:13]: Slight preference for that, yes, which would give this a less Android-specific flavour.

File: repositories.yaml (hunk 2 above)
Comment (Dexterp37) [reply repositories.yaml:28]: Oh, shame on me for not reading the commit message. Sorry!

**Review by Dexterp37 — APPROVED** [review]

**Pull request merged**